foundationdb/tests/timekeeper.rs

```diff
@@ -0,0 +1,78 @@
+use foundationdb::timekeeper::{hint_version_from_timestamp, HintMode};
+use std::cmp::{max, min};
+use std::time::SystemTime;
+
+#[tokio::test]
+async fn timekeeper() {
+    let _guard = unsafe { foundationdb::boot() };
+    let database = foundationdb::Database::new_compat(None)
+        .await
+        .expect("Unable to create database");
+    let trx_read = database.create_trx().expect("Unable to create transaction");
+    let now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("Unable to get timestamp")
+        .as_secs();
+    let read_version = trx_read
+        .get_read_version()
+        .await
+        .expect("Unable to get read version") as u64;
+    // A new transaction is needed because the one which get the read version has no
+    // knowledge about the new System Namespace Keyspace thus, further timekeeper
+    // keys are unknown for the first transaction.
+    // Creating a new one, make the first to be commited. The second will have the right
+    // timekeeper state.
+    let trx = database.create_trx().expect("Unable to create transaction");
+    tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+    let result = hint_version_from_timestamp(&trx, now, HintMode::AfterTimestamp, true)
+        .await
+        .expect("Unable to get hint version");
+
+    if let Some(hint_version) = result {
```

Comment:
suggestion: as this is experimental API, we should only check that `hint_version` > read_version, which should the case after the sleep, which we can extends a bit just-in-case.

Reply:
Empirically, it isn't always the case, sometime the timekeeper is to slow or too fast to write its keys.

That's why i have created this 20M range read version span. 

```
[foundationdb/tests/timekeeper.rs:35:9] hint_version = 3697606152912
[foundationdb/tests/timekeeper.rs:36:9] read_version = 3697606215326
```

Reply:
Either we wait a bit more than 10s, or we skip this verification, whatever you prefer.

Reply:
I've replaced it with the `assert!(matches!(Some(_), ...)`